src/main/java/com/chargerrobotics/subsystems/DriveSubsystem.java

```diff
@@ -68,6 +96,43 @@ public DriveSubsystem(XboxController driveController) {
 
     differentialDrive = new DifferentialDrive(leftDriveGroup, rightDriveGroup);
     differentialDrive.setDeadband(0.0);
+
+    odometryInitialized = false;
+
+    resetOdometry();
+
+    if (RobotBase.isSimulation()) {
+      fakeLeftEncoder = new Encoder(1, 2);
```

Comment:
Can you get these from the motor controller simulations? (I believe that the `CANSparkMax` classes do the simulation for themselves internally, though not 100% sure)

Reply:
@skallaher I might need some help with this. As far as I know, an `EncoderSim` needs an `Encoder` object passed into its constructor. Then the simulated hardware updates its readings based on numbers from the `drivetrainSim`. Updating the simulated hardware allows the `Encoder`'s `getDistance()` method to help update the simulated robot's pose. I don't know how to add `CANSparkMax` classes to the simulation because getting the encoder from a `CANSparkMax` class returns a `CANEncoder` object. My plan is to use `Encoder` and `EncoderSim` objects, but make their characteristics as close as possible to what's on the robot in real life.

Reply:
Hmmm... That does make things a bit more annoying. It must be the Talons that do this work for you.